hibernate-core/src/main/java/org/hibernate/boot/registry/classloading/internal/AggregatedClassLoader.java

```diff
@@ -196,18 +196,21 @@ protected URL findResource(String name) {
 	@Override
 	protected Class<?> findClass(String name) throws ClassNotFoundException {
 		final Iterator<ClassLoader> clIterator = newClassLoaderIterator();
+		Throwable t = new Throwable();
```

Comment:
You don't technically need to instantiate that throwable; see suggestion here and below.

```suggestion
		Throwable t = null;
```

Reply:
> I don't think having a Throwable around for the duration of the method would impact much in terms of memory and such?

It's more the call to `java.lang.Throwable#fillInStackTrace()` that annoys me, but yeah, I have no proof it's a problem, it's quite possibly just cargo cult.

Your PR, your rules :shrug: Feel free to merge :)

Reply:
Either way, eagerly creating the exception should be avoided.  We are creating it even if everything goes fine iiuc

Reply:
Yeah, I agree, Yoann's remark on the call to fillInStackTrace made me reconsider